### DISCUSSION TITLE
Seeking clarification on the role and necessity of `@tailwind components`

### DISCUSSION THREAD

**Discussion by guesung** [tailwindlabs/tailwindcss#14083] [Help] [answered]
While reviewing the Tailwind CSS source code, I noticed that `@tailwind base` and `@tailwind utilities` are exported in the [index.css](https://github.com/tailwindlabs/tailwindcss/blob/next/packages/tailwindcss/index.css) file. However, I couldn't find a clear reference to `@tailwind components`. 

After some experimentation, I observed no apparent difference when including or omitting `@tailwind components` in my projects. This raises questions about its purpose and implementation.

Could we provide more clarity on the role of `@tailwind components` and its impact on Tailwind CSS projects? It would be helpful to understand if this directive is still necessary or if its functionality has been incorporated elsewhere in the framework.

**Comment by ImEins** [comment, marked as answer]
Hello, the difference is pretty much clear, I quote the docs :

- The base layer is for things like reset rules or default styles applied to plain HTML elements.

**- The components layer is for class-based styles that you want to be able to override with utilities.**

- The utilities layer is for small, single-purpose classes that should always take precedence over any other styles.

They also added an example
```
@layer components {
  .card {
    background-color: theme('colors.white');
    border-radius: theme('borderRadius.lg');
    padding: theme('spacing.6');
    box-shadow: theme('boxShadow.xl');
  }
  /* ... */
}
```

Then we can use it like so :
```
<!-- Will look like a card, but with square corners -->
<div class="card rounded-none">
  <!-- ... -->
</div>
```

I'm not sure if this explains why the components layer exist..
Forgot to add the source : https://tailwindcss.com/docs/adding-custom-styles

**Reply by guesung** [reply]
Thank you for answering.

The components layer is for making custom styles! It's a very interesting feature. 

I missed that documents. Thank you.
